src/foam/u2/wizard/analytics/AlternateFlowRefinement.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * @license
+ * Copyright 2022 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.u2.wizard.analytics',
+  name: 'AlternateFlowRefinement',
+  refines: 'foam.u2.wizard.AlternateFlow',
+
+  imports: [
+    'analyticsAgent?'
+  ],
+
+  properties: [
+    {
+      class: 'Boolean',
+      name: 'enableAnalytics'
+    },
+    {
+      class: 'String',
+      name: 'analyticsName'
+    }
+  ],
+
+  methods: [
+    {
+      name: 'refinedMethod_execute_AlternateFlowRefinement',
+      code: foam.u2.wizard.AlternateFlow.getAxiomByName('execute').code
+    },
+    function execute (x) {
+      console.log('what is this', this);
+      if ( this.enableAnalytics && this.analyticsAgent ) {
+        const analyticsEvent = {
+          name: `AlternateFlow_${this.analyticsName}`,
+          tags: ['wizard']
+        };
+        this.analyticsAgent.pub('event', analyticsEvent);
+      } else {
+        console.warn(
+          'analytics is on for this AlternateFlow, but analytics is disabled',
+          this
+        );
+      }
```

Comment:
I meant this to log if an AlternateFlow is configured to send analytics, but it's in a context (a wizard) where analytics has not been enabled (there's no analyticsAgent to handle the event)

Reply:
oh if it's NOT configured to send analytics, do we still want to console.warn?

Reply:
If the analyticsAgent wasn't in context it would be hard to tell that this isn't working without manually testing. The situation where an alternate flow with analytics enabled is used in a context without analytics on purpose would be rare.

Reply:
on line 33 `if ( this.enableAnalytics && this.analyticsAgent ) {` because `enableAnalytics` is false by default, does it mean that we want to console.warn by default?

Reply:
No; fixed